gobblin-utility/src/main/java/gobblin/util/ProxiedFileSystemCache.java

```diff
@@ -134,16 +149,179 @@ public static FileSystem getProxiedFileSystemUsingKeytab(State state, URI fsURI,
   }
 
   /**
+   * Cached version of {@link ProxiedFileSystemUtils#createProxiedFileSystemUsingKeytab(String, String, Path, URI, Configuration)}.
+   * @deprecated use {@link #fromKeytab}.
+   */
+  @Deprecated
+  public static FileSystem getProxiedFileSystemUsingKeytab(@NonNull final String userNameToProxyAs,
+      final String superUserName, final Path superUserKeytabLocation, final URI fsURI, final Configuration conf)
+      throws ExecutionException {
+    try {
+      return getProxiedFileSystemUsingKeytab(userNameToProxyAs, superUserName, superUserKeytabLocation, fsURI, conf,
+          null);
+    } catch(IOException ioe) {
+      throw new ExecutionException(ioe);
+    }
+  }
+
+  /**
+   * Cached version of {@link ProxiedFileSystemUtils#createProxiedFileSystemUsingKeytab(String, String, Path, URI, Configuration)}.
+   */
+  @Builder(builderClassName = "ProxiedFileSystemFromKeytab", builderMethodName = "fromKeytab")
+  public static FileSystem getProxiedFileSystemUsingKeytab(@NonNull final String userNameToProxyAs,
+      final String superUserName, final Path superUserKeytabLocation, final URI fsURI,
+      final Configuration conf, FileSystem referenceFS) throws IOException, ExecutionException {
+    Preconditions.checkNotNull(userNameToProxyAs, "Must provide a user name to proxy as.");
+    Preconditions.checkNotNull(superUserName, "Must provide a super user name.");
+    Preconditions.checkNotNull(superUserKeytabLocation, "Must provide a keytab location.");
+    URI actualURI = resolveUri(fsURI, conf, referenceFS);
+    Configuration actualConfiguration = resolveConfiguration(conf, referenceFS);
+
+    return USER_NAME_TO_FILESYSTEM_CACHE.get(getFileSystemKey(actualURI, userNameToProxyAs, referenceFS),
+        new CreateProxiedFileSystemFromKeytab(userNameToProxyAs, superUserName, superUserKeytabLocation,
+            actualURI, actualConfiguration, referenceFS));
+  }
+
+  /**
    * Cached version of {@link ProxiedFileSystemUtils#createProxiedFileSystemUsingToken(String, Token, URI, Configuration)}.
+   * @deprecated use {@link #fromToken}.
    */
+  @Deprecated
   public static FileSystem getProxiedFileSystemUsingToken(@NonNull final String userNameToProxyAs,
       final Token<?> userNameToken, final URI fsURI, final Configuration conf) throws ExecutionException {
+    try {
+      return getProxiedFileSystemUsingToken(userNameToProxyAs, userNameToken, fsURI, conf, null);
+    } catch(IOException ioe) {
+      throw new ExecutionException(ioe);
+    }
+  }
+
+  /**
+   * Cached version of {@link ProxiedFileSystemUtils#createProxiedFileSystemUsingToken(String, Token, URI, Configuration)}.
+   */
+  @Builder(builderClassName = "ProxiedFileSystemFromToken", builderMethodName = "fromToken")
```

Comment:
Is there any advantage to deprecate previous methods and using builder here? The previous methods can also call  `decorateFilesystemFromReferenceFS` to get a `ratecontrolledFs`.